examples/erc20_gas/src/main.rs

```diff
@@ -0,0 +1,198 @@
+    let sender_balance = token_account
+        .storage
+        .get(&sender_balance_slot)
+        .expect("Balance slot not found")
+        .present_value();
+
+    if sender_balance < amount {
+        return Err(EVMError::Transaction(
+            InvalidTransaction::MaxFeePerBlobGasNotSupported,
+        ));
+    }
+    // Subtract the amount from the sender's balance
+    let sender_new_balance = sender_balance.saturating_sub(amount);
+    token_account.storage.insert(
+        sender_balance_slot,
+        EvmStorageSlot::new_changed(sender_balance, sender_new_balance),
+    );
+
+    // Add the amount to the recipient's balance
+    let recipient_balance_slot: U256 = keccak256((recipient, U256::from(3)).abi_encode()).into();
+    let recipient_balance = token_account
+        .storage
+        .get(&recipient_balance_slot)
+        .expect("To balance slot not found")
+        .present_value();
+    let recipient_new_balance = recipient_balance.saturating_add(amount);
+    token_account.storage.insert(
+        recipient_balance_slot,
+        EvmStorageSlot::new_changed(recipient_balance, recipient_new_balance),
+    );
+
+    Ok(())
+}
+
+fn balance_of(token: Address, address: Address, alloy_db: &mut AlloyCacheDB) -> Result<U256> {
+    sol! {
+        function balanceOf(address account) public returns (uint256);
+    }
+
+    let encoded = balanceOfCall { account: address }.abi_encode();
+
+    let mut evm = MainEvm::new(
+        Context::builder()
+            .with_db(alloy_db)
+            .modify_tx_chained(|tx| {
+                // 0x1 because calling USDC proxy from zero address fails
+                tx.caller = address!("0000000000000000000000000000000000000001");
+                tx.transact_to = TxKind::Call(token);
+                tx.data = encoded.into();
+                tx.value = U256::from(0);
+            }),
+        EthHandler::default(),
+    );
+
+    let ref_tx = evm.exec().unwrap();
+    let result = ref_tx.result;
+
+    let value = match result {
+        ExecutionResult::Success {
+            output: Output::Call(value),
+            ..
+        } => value,
+        result => return Err(anyhow!("'balanceOf' execution failed: {result:?}")),
+    };
+
+    let balance = <U256>::abi_decode(&value, false)?;
+
+    Ok(balance)
+}
+
+fn transfer(
+    from: Address,
+    to: Address,
+    amount: U256,
+    token: Address,
+    cache_db: &mut AlloyCacheDB,
+) -> Result<()> {
+    sol! {
+        function transfer(address to, uint amount) external returns (bool);
+    }
+
+    let encoded = transferCall { to, amount }.abi_encode();
+
+    let mut evm = MainEvm::new(
+        Context::builder()
+            .with_db(cache_db)
+            .modify_tx_chained(|tx| {
+                tx.caller = from;
+                tx.transact_to = TxKind::Call(token);
+                tx.data = encoded.into();
+                tx.value = U256::from(0);
+            }),
+        EthHandler::default(),
```

Comment:
@rakita So when I try to plug the custom handlers, it does not let me. As it expects the original handlers here with all the trait bounds. But the ones I have implemented also satisfies the trait bounds.

Reply:
Where is the implementation of your Handler?

Reply:
Here: https://github.com/royvardhan/revm/tree/1782-1/examples/erc20_gas/src/handlers

Reply:
Will need to check errors

Reply:
I am trying to do this: 
![image](https://github.com/user-attachments/assets/795f32b9-e5af-4139-bf32-720b2046e0b1)

Reply:
One of the errors: ```
arguments to this function are incorrect
expected struct `EthValidation<Context<BlockEnv, TxEnv, CfgEnv, &mut CacheDB<WrapDatabaseAsync<AlloyDB<..., ..., ...>>>>, ...>`
   found struct `Erc20Validation`
```

Reply:
Hm it seems that EthHandler::new needs to be changed slightly.

Reply:
There is no `new` method in `[Handler](https://github.com/bluealloy/revm/blob/45581c451a440776fd59576d7b27c366b1528724/crates/handler/interface/src/handler.rs#L3)`. So I did this, still not supporting: 
<img width="928" alt="image" src="https://github.com/user-attachments/assets/dcc603a4-2e04-4d8a-af48-7f19a66a43c9">

Reply:
Due to this [MainEvm](https://github.com/bluealloy/revm/blob/45581c451a440776fd59576d7b27c366b1528724/crates/revm/src/evm.rs#L150) it is taking the default handler as `EthHandler`

Reply:
did you try creating a type alias for your handler like its done for op here https://github.com/bluealloy/revm/blob/45581c451a440776fd59576d7b27c366b1528724/crates/optimism/src/handler.rs#L40

and then do a CustomEvm type alias similar to MainEvm but replaces the default Ethhandlert.
and then go CustomEvm::new(..

Reply:
@FredCoen thanks man, fixed!

Reply:
awesome 💪💪

Reply:
Awesome. I will check it tomorrow